modules/apps/remote-js-script/remote-js-script-admin-web/src/main/java/com/liferay/remote/js/script/admin/web/internal/portlet/RemoteJsScriptPortlet.java

```diff
@@ -57,36 +61,67 @@ public String getName(Locale locale) {
 	}
 
 	public synchronized void register(BundleContext bundleContext) {
-		if (_serviceRegistration != null) {
-			throw new IllegalStateException("Portlet is already registered");
-		}
-
-		Dictionary<String, Object> properties = new Hashtable<>();
-
-		properties.put(
-			"com.liferay.portlet.css-class-wrapper", "portlet-remote-js-script");
-		properties.put(
-			"com.liferay.portlet.display-category", "category.sample");
-		properties.put(
-			"com.liferay.portlet.header-portlet-css", "/display/css/main.css");
-		properties.put("com.liferay.portlet.instanceable", true);
-		properties.put("javax.portlet.name", _getPortletName());
-		properties.put("javax.portlet.security-role-ref", "power-user,user");
-		properties.put(
-			"javax.portlet.resource-bundle", _getResourceBundleName());
-
-		_serviceRegistration = bundleContext.registerService(
-			Portlet.class, this, properties);
+		String customElementName = _remoteJsScriptEntry.getCustomElementName();
 
-		properties = new Hashtable<>();
-
-		properties.put("resource.bundle.base.name", _getResourceBundleName());
-		properties.put("servlet.context.name", "remote-js-script-admin-web");
+		if (customElementName != null) {
+			if (_dynamicIncludeServiceRegistration != null) {
+				throw new IllegalStateException("DynamicInclude is already registered");
+			}
 
-		_resourceBundleLoaderServiceRegistration =
-			bundleContext.registerService(
-				ResourceBundleLoader.class,
-				locale -> _getResourceBundle(locale), properties);
+			_dynamicIncludeServiceRegistration = bundleContext.registerService(
```

Comment:
Is this registering a new DynamicInclude per script? Maybe it would be better to register just one, and generate all the `<script>`  tags from it... :thinking:

Reply:
Maybe not, since there's only one `_dynamicIncludeServiceRegistration`... Still I don't see why we need to do the registration dinamically: we can register a DynamicInclude statically, then generate its contents dynamically, I think...